File: ProgrammingBasicsJava/Jan2020/NestedLoopsExercise/src/_03SumPrimeNonPrime.java
import java.util.Scanner;

public class _03SumPrimeNonPrime {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        String command = scanner.nextLine();

        int sumPrimeNumbers = 0;
        int sumNonPrimeNumbers = 0;

        while (!"stop".equals(command)) {
            int currentNum = Integer.parseInt(command);
            boolean isNumberPrime = true;

            if (currentNum < 0) {
                System.out.println("Number is negative.");
            }
            else {
                if (currentNum == 0 || currentNum == 1) {
                    sumNonPrimeNumbers += currentNum;
                }
                else {
                    for (int i = 2; i < currentNum; i++) {
                        if (currentNum % i == 0) {
                            sumNonPrimeNumbers += currentNum;
                            isNumberPrime = false;
                            break;
                        }
                    }
                    if (isNumberPrime) {
                        sumPrimeNumbers += currentNum;
                    }
                }
            }

            command = scanner.nextLine();
        }

        System.out.printf("Sum of all prime numbers is: %d%n", sumPrimeNumbers);
        System.out.printf("Sum of all non prime numbers is: %d%n", sumNonPrimeNumbers);
    }
}

File: JavaScriptBasics/06_2_NestedLoopsExercise/03_SumPrimeNonPrime.js
function printSumPrimeNonPrimeNumbers(input) {
    let index = 0;
    let command = input[index];
    let sumPrimeNumbers = 0;
    let sumNonNumbers = 0;

    while (command !== "stop") {
        let currentNum = Number(command);

        if (currentNum < 0 ) {
            console.log("Number is negative.");
        } else {
            let isPrime = true;

            for (i = 2; i < currentNum; i++) {
                if (currentNum % i == 0) {
                    isPrime = false;
                    break;
                }
            }

            if (currentNum > 1 && (currentNum === 2 || isPrime)) {
                sumPrimeNumbers += currentNum;
            } else {
                sumNonNumbers += currentNum;
            }
        }

        index++;
        command = input[index];
    }

    console.log(`Sum of all prime numbers is: ${sumPrimeNumbers}`);
    console.log(`Sum of all non prime numbers is: ${sumNonNumbers}`);
}

printSumPrimeNonPrimeNumbers(["3",
"9",
"0",
"7",
"19",
"4",
"stop"]);